dali/operators/geometry/affine_transforms/rotate_transform.cc

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/geometry/affine_transforms/transform_base_op.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/core/geom/transform.h"
+#include "dali/core/math_util.h"
+
+namespace dali {
+
+DALI_SCHEMA(RotateTransform)
+  .DocStr(R"code(Produces a rotation affine transform matrix.
+
+If another transform matrix is passed as an input, the operator applies rotation to the matrix provided.
+
+The number of dimensions will be 2 or 3 depending on whether the argument ``axis`` is provided or not,
+respectively.
```

Comment:
The "respectively" makes it the other way round:
"2 if provided, 3 if not".
Maybe let's shorten it:
```suggestion
The number of dimensions is assumed to be 3 if a rotation axis is provided or 2 otherwise.
```